File: backend/src/main/java/pl/michalkruk/cinema/core/movie/Movie.java
package pl.michalkruk.cinema.core.movie;

import lombok.*;

import javax.persistence.*;

@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Movie {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Enumerated(EnumType.STRING)
    private Genre genre;

    @Enumerated(EnumType.STRING)
    private AgeLimit ageLimit;

    private short duration;

    private short releaseYear;

    @Enumerated(EnumType.STRING)
    private Country country;

    @Column(length = 2000)
    private String description;

    private String imageName;
}

File: frontend/src/main/web/src/app/movie/movie.ts
import {Genre} from "./genre";
import {Country} from "./contry";
import {Show} from "./show";
import {AgeLimit} from "./age-limit";

export class Movie {
  constructor(
    public id: number,
    public title: string,
    public genre: Genre,
    public ageLimit: AgeLimit,
    public duration: number,
    public releaseYear: number,
    public country: Country,
    public description: string,
    public image: any,
    public shows: Show[]
  ) {
  }
}
